ksql-parser/src/main/java/io/confluent/ksql/util/DataSourceExtractor.java

```diff
@@ -208,6 +136,81 @@ public Schema getFromSchema() {
     return fromSchema;
   }
 
+  private final class Visitor extends SqlBaseBaseVisitor<Node> {
+
+    @Override
+    public Node visitQuerySpecification(final SqlBaseParser.QuerySpecificationContext ctx) {
+      visit(ctx.from);
+      return visitChildren(ctx);
+    }
+
+    @Override
+    public Node visitTableName(final SqlBaseParser.TableNameContext context) {
+      return new Table(getLocation(context), getQualifiedName(context.qualifiedName()));
+    }
+
+    @Override
+    public Node visitAliasedRelation(final SqlBaseParser.AliasedRelationContext context) {
+      final Table table = (Table) visit(context.relationPrimary());
+
+      String alias = null;
+      if (context.children.size() == 1) {
+        alias = table.getName().getSuffix().toUpperCase();
+
+      } else if (context.children.size() == 2) {
+        alias = context.children.get(1).getText().toUpperCase();
+      }
+
+      if (!isJoin) {
+        fromAlias = alias;
+        fromName = table.getName().getSuffix().toUpperCase();
+        final StructuredDataSource
+            fromDataSource =
+            metaStore.getSource(table.getName().getSuffix());
+        if (fromDataSource == null) {
+          throw new KsqlException(table.getName().getSuffix() + " does not exist.");
+        }
+        fromSchema = fromDataSource.getSchema();
+        return null;
+      }
+
+      // TODO: Figure out if the call to toUpperCase() here is really necessary
+      return new AliasedRelation(getLocation(context), table, alias.toUpperCase(),
```

Comment:
potential NPE? `alias` could be null based on the `if...else if`

Reply:
You're commenting on existing code. All I've done is move it.

Reply:
If there are no tests then perhaps the PR should include some?

Reply:
I've deliberately not changed any functionality here because there are no tests.  If I have to write tests then I'll have to close the PR as I don't have the time to invest in working out what this class is doing to the level that I can write tests.